main.tf

```diff
@@ -118,25 +129,24 @@ data "aws_iam_policy_document" "bucket_policy" {
   }
 
   statement {
+    sid = "cloudtrail-logs-put-object"
+
     actions = ["s3:PutObject"]
-    effect  = "${(var.default_allow || var.allow_cloudtrail) ? "Allow" : "Deny"}"
 
-    condition {
-      test     = "StringEquals"
-      variable = "s3:x-amz-acl"
-      values   = ["bucket-owner-full-control"]
-    }
+    effect = "${(var.default_allow || var.allow_cloudtrail) ? "Allow" : "Deny"}"
 
     principals {
       type        = "Service"
       identifiers = ["cloudtrail.amazonaws.com"]
     }
 
-    resources = [
-      "arn:aws:s3:::${var.s3_bucket_name}/${var.cloudtrail_logs_prefix}/AWSLogs/${data.aws_caller_identity.current.account_id}/*",
-    ]
+    resources = "${length(var.cloudtrail_accounts) > 0 ? formatlist(format("arn:aws:s3:::%s/%s/AWSLogs/%%s/*",var.s3_bucket_name, var.cloudtrail_logs_prefix), var.cloudtrail_accounts) : list(format("arn:aws:s3:::%s/%s/AWSLogs/%s/*", var.s3_bucket_name, var.cloudtrail_logs_prefix, data.aws_caller_identity.current.account_id))}"
```

Comment:
```
arn:aws:s3:::%s/%s/AWSLogs/%%s/*
```

Why is the %%s necessary? for the last part? Wouldn't that just make it a literal '%s' rather than interpolating the value?

Reply:
Oh I think I see, that then gets passed to the formatlist function to pass in the cloudtrail_accounts.